src/codeview.js

```diff
@@ -158,6 +164,27 @@ var Codeview = GObject.registerClass({
         this._helpMessage.popup();
     }
 
+    static _onInsertText(buffer, location, text) {
+        const sound = SoundServer.getDefault();
+        [...text].forEach(letter => {
```

Comment:
I feel that you could simplify this by having a dictionary mapping letter -> sound name. Then you can look at `dictionary[letter]` and play that if it exists, without having to enumerate every case in the switch.